src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -123,6 +123,8 @@
     /** Current limit for this particular connection. */
     private Boolean enablePrepareOnFirstPreparedStatementCall = null;
```

Comment:
why does this need to be Boolean?

Reply:
So that is related to this feature here: https://github.com/microsoft/mssql-jdbc/wiki/PreparedStatement-metadata-caching

It's kinda confusing because it uses the "prepare" wording. So if that is enabled and depending on the prepareMethod, the driver will call sp_prepexec or sp_prepare  -> sp_execute for the first execution